fluent-syntax/src/parser.js

```diff
@@ -766,14 +668,44 @@ export default class FluentParser {
     if (ps.currentChar === "-") {
       ps.next();
       const id = this.getIdentifier(ps);
-      return new AST.TermReference(id);
+
+      let attr;
+      if (ps.currentChar === ".") {
+        ps.next();
+        attr = this.getIdentifier(ps);
+      }
+
+      let args;
+      if (ps.currentChar === "(") {
+        args = this.getCallArguments(ps);
+      }
+
+      return new AST.TermReference(id, attr, args);
     }
 
     if (ps.isIdentifierStart()) {
       const id = this.getIdentifier(ps);
-      return new AST.MessageReference(id);
+
+      if (ps.currentChar === "(") {
+        // It's a Function. Ensure it's all upper-case.
+        if (!/^[A-Z][A-Z_?-]*$/.test(id.name)) {
```

Comment:
this looks weird... I'd expect that function identifier be a subset of EBNF identifier which is `[a-zA-Z] [a-zA-Z0-9_-]*` and I'd expect the function to be `[A-Z] [A-Z0-9_-]*`

Instead you cut out numbers and added `?` which I think is a leak of your old idea to use `?` in function names. Can you clarify what's the spec here?

Reply:
This wasn't intentional. I probably kept copying this code for the few last releases from one place to another. The spec has it as `const VALID_FUNCTION_NAME = /^[A-Z][A-Z0-9_?-]*$/;` which also shouldn't include the `?`. I'll fix it.

Reply:
oh, cool! thanks. It would be also nice to have those things tested in fixtures (`_`, `-` and `0-9`)

Reply:
I fixed this in the spec in https://github.com/projectfluent/fluent/issues/244 and I've since updated this PR.